apps/files_sharing/lib/Controller/ShareAPIController.php

```diff
@@ -199,6 +201,26 @@ protected function formatShare(IShare $share, ?Node $recipientNode = null): arra
 		$result['item_size'] = $node->getSize();
 		$result['item_mtime'] = $node->getMTime();
 
+		if ($this->trustedServers !== null && in_array($share->getShareType(), [IShare::TYPE_REMOTE, IShare::TYPE_REMOTE_GROUP])) {
+			$result['is_trusted_server'] = false;
+			$sharedWith = $share->getSharedWith();
+			$remoteIdentifier = is_string($sharedWith) ? strrchr($sharedWith, '@') : false;
+			if ($remoteIdentifier !== false) {
+				$remote = substr($remoteIdentifier, 1);
+				try {
+					if ($this->trustedServers->isTrustedServer($remote)) {
+						$result['is_trusted_server'] = true;
+					}
+				} catch (\Exception $e) {
+					// Server not found or other issue, we consider it not trusted
+					$this->logger->error(
+						'Error checking if remote server is trusted (treating as untrusted): ' . $e->getMessage(),
+						['exception' => $e]
+					);
```

Comment:
```suggestion
					$result['is_trusted_server'] = false;
					$this->logger->error(
						'Error checking if remote server is trusted (treating as untrusted): ' . $e->getMessage(),
						['exception' => $e]
					);
```

Set it to false explicitly.

Reply:
It is set earlier in the block.